fuseimg/datasets/picai.py

```diff
@@ -0,0 +1,284 @@
+from fuse.data.datasets.dataset_default import DatasetDefault
+from fuse.data.datasets.caching.samples_cacher import SamplesCacher
+from fuseimg.data.ops.color import OpNormalizeAgainstSelf
+
+from fuseimg.data.ops.aug.geometry import OpAugAffine2D, OpAugSqueeze3Dto2D, OpAugUnsqueeze3DFrom2D
+from fuse.data import PipelineDefault, OpToTensor, OpRepeat
+from fuse.data.ops.ops_common import OpLambda, OpLookup, OpToOneHot
+
+from fuse.data.ops.ops_aug_common import OpRandApply, OpSampleAndRepeat
+from fuse.data.ops.ops_read import OpReadDataframe
+from fuse.data.ops.ops_cast import OpToNumpy
+from fuse.data.ops.op_base import OpBase
+from fuse.utils import NDict
+from functools import partial
+import nibabel as nib
+from typing import Hashable, Optional, Sequence
+import torch
+import pandas as pd
+import numpy as np
+import skimage
+import os
+from fuse.data.utils.sample import get_sample_id
+from medpy.io import load
+
+from fuse.utils.rand.param_sampler import Uniform, RandInt
+
+
+class OpPICAISampleIDDecode(OpBase):
+    """
+    decodes sample id into image and segmentation filename
+    """
+
+    def __call__(self, sample_dict: NDict) -> NDict:
+        """ """
+        sid = get_sample_id(sample_dict)
+
+        img_filename_key = "data.input.img_path"
+        sample_dict[img_filename_key] = sid
+
+        return sample_dict
+
+
+class OpLoadPICAIImage(OpBase):
+    """
+    Loads a medical image
+    """
+
+    def __init__(self, dir_path: str, seuqences: Sequence[str] = ["_t2w"], **kwargs: dict):
+        super().__init__(**kwargs)
+        self._dir_path = dir_path
+        self._sequences = seuqences
+
+    def __call__(
+        self,
+        sample_dict: NDict,
+        key_in: str,
+        key_out: str,
+    ) -> None:
+        """
+        :param key_in: the key name in sample_dict that holds the filename
+        :param key_out: the key name in sample_dict that holds the image
+        :param key_metadata_out : the key to hold metadata dictionary
+        """
+        try:
+            for seq in self._sequences:
+                img_filename = os.path.join(
+                    self._dir_path, sample_dict[key_in].split("_")[0], sample_dict[key_in] + seq + ".mha"
+                )
+
+                image_data, image_header = load(img_filename)
+                sample_dict[key_out + seq] = image_data
+            return sample_dict
+        except:
+            return None
```

Comment:
maybe you should display a clearer warning when this happens?

Reply:
what do you mean by clearer?
@mosheraboh any idea?